network/peers/conn_manager.go

```diff
@@ -192,3 +208,60 @@ func scorePeer(peerSubnets records.Subnets, subnetsScores []float64) PeerScore {
 	}
 	return PeerScore(score)
 }
+
+// DisconnectFromBadPeers will disconnect from bad peers according to their GossipSub scores. It returns the number of disconnected peers.
+func (c connManager) DisconnectFromBadPeers(logger *zap.Logger, net libp2pnetwork.Network, allPeers []peer.ID) int {
+
+	disconnectedPeers := 0
+
+	for _, peerID := range allPeers {
+
+		// Check if peer is bad
+		if c.gossipSubScoreIndex.HasBadGossipSubScore(logger, peerID) {
+
+			// Get GossipSub score for logging
+			gossipSubScore, hasScore := c.gossipSubScoreIndex.GetGossipSubScore(peerID)
```

Comment:
I think this could be better as one call. 
So something like  `gossipSubScore, hasScore := c.gossipSubScoreIndex.GetGossipSubScore(peerID)` and then calculate if `gossipSubScore` is over a threshold for being bad. 
not too bad but I don't see a reason searching the peer in the map twice..

Reply:
Indeed, I'll fix that
Thanks